### ISSUE TITLE
Brave instantly crashes upon attempting to open any Youtube tab

### ISSUE THREAD

**Issue by RougeNPS** [brave/brave-browser#37243]
## Description 

Brave will instantly crash upon opening any Youtube tab, regardless if it is the homepage, a video, channel page, etc. Started 2:52 PM EST. Yes i am on Windows 8, no i dont have access or option to upgrade to Windows 10. 

## Steps to Reproduce

   1.  Open Brave.
   2.  Open Youtube.
   3. 

## Actual result:

Crash.

## Expected result:
Youtube opens/works.

## Reproduces how often: 

Every time.

## Brave version (brave://version info)

1.47.186 Chromium: 109.0.5414.119 (Official Build) (32-bit) 
772095164c7d5d4e73160f858efed3b5e87eca83-refs/branch-heads/5414@{#1458}
--
OS | Windows 8.1 (Build 9600.20371)

## Version/Channel Information:

- Can you reproduce this issue with the current release? 
- Can you reproduce this issue with the beta channel? 
- Can you reproduce this issue with the nightly channel? 

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields?  No, tested multiple settings.
- Does the issue resolve itself when disabling Brave Rewards? No, always off.
- Is the issue reproducible on the latest version of Chrome? Unable to test.

## Miscellaneous Information:

Crashes with videos played in sites like Discord as well.

**Comment by 77ro** [comment]
I'm using 7 pro and Brave and also unable to play videos- issue starting today.

**Comment by bew-dev** [comment]
same here, and just started happening today on Brave.  Youtube was in the tabs, and when restore button was clicked  i couldnt close tabs in time, finally i got youtube tabs closed and it opens now.  then as soon as i open youtube it closes right away. 

I disabled ublock origin, and 2 other youtube extensions and it didnt help. 
I have BitWarden, OneTab, and StartPage extensions enabled only.

**Comment by getsmass** [comment]
Same here. Did not even reopen Brave. Had a session open for a few weeks. 4 YT dormant tabs. Switched to one of them, and Brave crashed. Can't restore session, as it immediately crashes. Blank session with only YT - crash. Was working fine yesterday. Win 7 Pro. 1.47.186.

**Comment by trollezt** [comment]
My Brave crashes even when I type the word "Youtube" in a Google search. When restoring tabs, there are a couple of seconds to close the problematic tab, otherwise the browser crashes again.

**Comment by Renkineko** [comment]
Same here, google crash randomly (mainly when there is pictures in the result, like in people search). Google can stop the crash sometimes if you deactivate brave shield. The youtube is a direct crash, even when deactivating shield. Linux ubuntu here, [Version 1.47.186 Chromium: 109.0.5414.119 (Build officiel) (64 bits)](https://brave.com/latest/)

Stacktrace when crashing on youtube:

```shell
$ RUST_BACKTRACE=full brave-browser
[1564946:1564946:0402/132247.707172:ERROR:profile_attributes_storage.cc(846)] Failed to PNG encode the image.
[1564946:1564978:0402/132247.873263:ERROR:nss_util.cc(357)] After loading Root Certs, loaded==false: NSS error code: -8018
[1564946:1565005:0402/132247.906229:ERROR:dat_file_util.cc(23)] GetDATFileData: the dat file is not found or corrupted /home/renkineko/.config/BraveSoftware/Brave-Browser/afalakplffnnnlkncjhbmahjfjhmlkal/1.0.682/1/ExtensionWhitelist.dat
[1564946:1564946:0402/132247.906417:ERROR:extension_whitelist_service.cc(73)] Could not obtain extension whitelist data
thread '<unnamed>' panicked at 'index out of bounds: the len is 9 but the index is 9', /home/ubuntu/workspace/brave-browser-build-linux-x64-release/src/brave/build/rustup/1.1.0/registry/src/github.com-1ecc6299db9ec823/adblock-0.6.2/src/resources/scriptlet_resource_storage.rs:44:25
stack backtrace:
   0:     0x55daffac8e1c - <unknown>
   1:     0x55daffaecbbc - <unknown>
   2:     0x55daffac6223 - <unknown>
   3:     0x55daffacb202 - <unknown>
   4:     0x55daffacade5 - <unknown>
   5:     0x55daffacb8b3 - <unknown>
   6:     0x55daffacb5d0 - <unknown>
   7:     0x55daffac92c4 - <unknown>
   8:     0x55daffacb309 - <unknown>
   9:     0x55daffaeaa81 - <unknown>
  10:     0x55daffaeaa42 - <unknown>
  11:     0x55daffa00651 - <unknown>
  12:     0x55daff9a4134 - <unknown>
  13:     0x55db05f8f075 - <unknown>
[0402/132252.147148:ERROR:elf_dynamic_array_reader.h(64)] tag not found
[1564985:1564991:0402/132258.335728:ERROR:broker_posix.cc(40)] Recvmsg error: Connexion ré-initialisée par le correspondant (104)
/usr/bin/brave-browser : ligne 48 : 1564946 Abandon                 (core dumped) "$HERE/brave" "$@"
```

Note: no folder /home/ubuntu exists. First 4 lines are always on startup. Crash begins with `thread '<unnamed>'` line.

First few lines are always displayed, even when I move the .config folder to start from fresh.

May update this message later with further tests.

**Comment by SoundcoDJ** [comment]
Same here as of 4/1/2024. I'm using a MacBook Pro running OS 10.14.6
YouTube crashes Brave within 10 seconds of attempting to open their site.

**Comment by theguywhoaskeed** [comment]
i found a solution go to settings privacy and security scroll down a little bit you'll find an option sites and shields click on it there will be an option shields and now just add youtube to shields down thats it. youtube will start working properly. youtube is blocking shields and crashing the browser for that you can ublock origin or something.
excuse my english

**Comment by ChaosCom** [comment]
I can confirm having the same issue on a Win7 Laptop (that I can't update to newer versions of Windows).

Some insights:

This has to be related to the built-in content filter engine, because turning "Brave Shields" off for YT explicitly by adding it to the "Shields Down" section of 

Settings -> Privacy and Security -> Site and Shields Settings -> Shields status

stops the browser crashes. (Of course, you can also disable "Shields" completely for ALL sites, but going through the path described above allows you to disable content filtering specifically to the affected sites; I also read that for some people the google search page also causes crashes)

On a side note, if you use the uBlock Origin extension and the filters through that extension (filter definitions are mostly the same between the brave-integrated content filter ("Shields") and uBlock Origin), then then content filtering will work for YT as it will occur through UO.

So narrowing it down to Brave's integrated content filter, it only can be one of the two:
1. Brave's content filter engine itself
2. Some content filter definition

What is really worrying is that the content filter engine brings down the *entire* browser, even with all the process-isolation for tabs and whatnot that takes place in modern chromium-based browsers. Which could make for a very nasty malware-entry point.

**Comment by theguywhoaskeed** [comment]
I've noticed the crashing problem is with all the google related services

**Comment by rebron** [comment]
cc: @ryanbr

**Comment by cdavis5e** [comment]
I can confirm that this is also occurring on macOS.

Brave 1.57.64 (Chromium 116.0.5845.188), macOS 10.14.6 (18G9323).

**Comment by sonderblue** [comment]
have tried to set shields down for youtube, brave continues to crash all the same

**Comment by ryanbr** [comment]
Yeah, its due to the older win7 adblock-rust code base.

- If you’re on Win 10/11/Linux/Mac. No changes needed, just use the latest version of Brave.

- If you’re on windows 7, 8, 8.1, disable (globally) sheilds via brave://settings/shields. “Ads & trackers” and install uBlock Origin via [https://chromewebstore.google.com/](https://chromewebstore.google.com/)

Crashes only occur since win7/8 builds aren’t maintained and don’t support the current adblock rules causing crashing.

**Comment by RougeNPS** [comment]
Thank you for providing an actual solution instead of just saying "Yeah update your OS." thats really appreciated. 

Should i close the issue or what?

**Comment by ryanbr** [comment]
Sure, lets close this. If you can't upgrade to the latest OS (and get the latest Brave ver). And still want to use Brave.

- Disable (globally) sheilds via brave://settings/shields. “Ads & trackers” 
- Install uBlock Origin via https://chromewebstore.google.com/
- From uBlock Origin enable `Fanboy Annoyances` or `Easylist Cookie lists`, for feature parity with Brave.

**Comment by astrostab** [comment]
Same issue here, win7 64bit, clean brave without any extensions or plugins, version 1.47.186 Chromium: 109.0.5414.119. Shields is on with standard settings. Worked perfectly for months no issues ever, session always active for weeks, after work few videos watched then PC hibernated. Youtube worked Thu as normal, crashes every time on Friday 5th Apr 2024. Any access to any youtube page, link or video, instant crash of entire brave browser with all tabs closed. Youtube works fine on firefox, search for youtube on duckduckgo works fine. 

From few other sites seems issue was happening on Win 7, Win 10, Android, Ubuntu linux and MacOS.

Today 6th Apr 2024 youtube works again with no issues. Nothing was changed on my end, same browser, same PC, no addons no extensions.
Only thing I did was change from standard to aggressive in brave "/settings/trackers and ads blocking", and back to standard and cleared the cache.

I bet it had little to do with brave but lots to do with youtube /google page scripts or they got hacked and someone injected some code. Brave is based on chromium same as Chrome so parts of it possibly written by google .... they may know some bugs in there and how to explore them...

On the other hand if some script or error on a single page can crash entire browser closing all tabs then its a junky low quality code in there used by millions. We have big bugs in the software we use everyday!!!

If I was using browsers for anything serious I would be worried.
